File: Kotlin/app/src/main/java/dk/coded/emia/View/pages/MyFavoritePostsFragment.java
package dk.coded.emia.View.pages;

import android.app.Activity;
import android.util.Log;
import dk.coded.emia.View.fragment.PostListFragment;
import dk.coded.emia.model.Data.Post;
import dk.coded.emia.model.Data.User;
import dk.coded.emia.model.adapter.PostListValidator;
import dk.coded.emia.utils.BasicCallBack;
import dk.coded.emia.model.interactor.DatabaseFactory;
import dk.coded.emia.model.interactor.DatabaseInteractor;
import static dk.coded.emia.utils.Constants.FAIL;
import static dk.coded.emia.utils.Constants.SUCCESS;

public class MyFavoritePostsFragment extends PostListFragment {

    private DatabaseInteractor mInteractor;

    public MyFavoritePostsFragment() {
        mInteractor = DatabaseFactory.INSTANCE.getDatabaseInteractor();
    }

    @Override
    public void needShow(Post post, User user, BasicCallBack callback) {

        Log.d(getClass().getName(), "handle: " + post.getId());

        if (!getMValidator().isValid(post, user)) {
            callback.callBack(Companion.getSUCCESS(), false);
        } else {
            mInteractor.isItMyFavoritePost(post, (int status, Object data) -> {
                if (status == Companion.getSUCCESS()) {
                    Boolean isFavorite = (Boolean) data;
                    Log.d(getClass().getName(), "Post " + post.getId() + " is my favorite - " + isFavorite.toString());
                    callback.callBack(Companion.getSUCCESS(), isFavorite);
                } else if (status == Companion.getFAIL()) {
                    callback.callBack(Companion.getSUCCESS(), false);
                }
            });
        }
    }
}

File: Kotlin/app/src/main/java/dk/coded/emia/View/pages/MyFavoritePostsFragment.kt
package dk.coded.emia.View.pages

import android.app.Activity
import android.util.Log
import dk.coded.emia.View.fragment.PostListFragment
import dk.coded.emia.model.Data.Post
import dk.coded.emia.model.Data.User
import dk.coded.emia.model.adapter.PostListValidator
import dk.coded.emia.utils.BasicCallBack
import dk.coded.emia.model.interactor.DatabaseFactory
import dk.coded.emia.model.interactor.DatabaseInteractor
import dk.coded.emia.utils.Constants

class MyFavoritePostsFragment : PostListFragment() {

    private val mInteractor: DatabaseInteractor

    init {
        mInteractor = DatabaseFactory.databaseInteractor
    }

    override fun needShow(post: Post, user: User, callback: BasicCallBack) {

        Log.d(javaClass.name, "handle: " + post.id!!)

        if ((mValidator!!.isValid(post, user))!! == false) {
            callback(Constants.SUCCESS, false)
        } else {
            mInteractor.isItMyFavoritePost(post) { status: Int, data: Any? ->
                if (status == Constants.SUCCESS) {
                    val isFavorite = data as Boolean
                    Log.d(javaClass.name, "Post " + post.id + " is my favorite - " + isFavorite.toString())
                    callback(Constants.SUCCESS, isFavorite)
                } else if (status == Constants.FAIL) {
                    callback(Constants.SUCCESS, false)
                }
            }
        }
    }
}
